popsycle/synthetic.py

```diff
@@ -3512,3 +3527,390 @@ def generate_slurm_scripts(slurm_config_filename, popsycle_config_filename,
+        Defaults to 8.3 kpc (Where Earth is located)
+
+    c : float
+        Concentration index.
+        Defaults to 10 (concentration index of the Milky Way, multiple papers show it to be anywhere from 10-12)
+
+    r_vir : float
+        The virial radius.
+        Defaults to 200 kpc (The virial radius of the Milky Way)
+
+    Optional Parameters
+    -------------------
+
+    overwrite : bool
+        If set to True, overwrites output files. If set to False, exists the
+        function if output files are already on disk.
+        Default is False.
+
+    seed : int
+        If set to non-None, all random sampling will be seeded with the
+        specified seed, forcing identical output for PyPopStar and PopSyCLE.
+        Default None.
+
+    Outputs
+    -------
+    <output_root2>.h5 : hdf5 file
+        The new .h5 file with PBHs injected in.
+    """
+    ##########
+    # Error handling: check whether files exist and
+    # whether input types are correct.
+    ##########
+
+    if not overwrite:
+        # Check if HDF5 file exists already. If it does, throw an error message
+        # to complain and exit.
+        if os.path.isfile(output_root2 + '.h5'):
+            raise Exception(
+                'That .h5 file name is taken! Either delete the .h5 file, '
+                'or pick a new name.')
+
+    # Error handling/complaining if input types are not right.
+
+    if type(output_root2) != str:
+        raise Exception('output_root must be a string.')
+
+    if type(galaxia_area) != float:
+        if type(galaxia_area) != int:
+            raise Exception('galaxia_area must be a float or an integer.')
+
+    if type(fdm) != float:
+        if type(fdm) != int:
+            raise Exception('fdm must be a float or an integer.')
+
+    if type(pbh_mass) != int:
+        if type(pbh_mass) != float:
+            raise Exception('pbh_mass must be an integer or a float.')
+
+    if type(r_max) != float:
+        if type(r_max) != int:
+            raise Exception('r_max must be a float or an integer.')
+
+    if type(c) != int:
+        if type(c) != float:
+            raise Exception('c must be an integer or a float.')
+
+    if type(r_vir) != float:
+        if type(r_vir) != int:
+            raise Exception('r_vir must be a float or an integer.')
+
+    if seed is not None:
+        if type(seed) != int:
+            raise Exception('seed must be an integer.')
+
+    ##########
+    # Start of code
+    #########
+
+    # Set random seed
+    np.random.seed(seed)
+
+    t0 = time.time()
+
+    #Define parameters for NFW profile calculations
+    r_s = r_vir/c #kpc, scale radius
+    g = 4.3*(10**-3) #(pc*km^2)/(Msun*s^2)
+    h = 70 #km/(s*Mpc)
+
+    #Read in the hdf5 file that doesn't have PBHs. Product of perform_pop_syn.
+    no_pbh_hdf5_file = h5py.File(hdf5_file)
+    key_list = list(no_pbh_hdf5_file)
+    key_list = key_list[:len(key_list)-2]
```

Comment:
Is this meant to remove the 'lat_bin_edges' and 'lon_bin_edges' keys? If so, then I would rather see something more explicit, like:
`key_list = [key for key in key_list if 'bin_edges' not in key]`